### PR TITLE
feat: add date picker to class modal

### PR DESCRIPTION
**Pull request by wais-hundekar**
## Notion ticket link

[Update Create Class modal to match updated designs](https://www.notion.so/uwblueprintexecs/47bf4727a2444bccad4a91952898b8f3?v=9658436bf9be409e90581631a7155f16&p=1c95e8134c4e4c63b8644b3d124ffc07&pm=s)

## Implementation description
* Updated backend to support `startDate` over `schoolYear`
* Implemented date picker from [chakra-dayzed-datepicker
](https://github.com/aboveyunhai/chakra-dayzed-datepicker)
* Resolved all resulting TypeScript errors and updated code to match designs
Screenshot 2023-05-23 at 10 07 21 PM

Screenshot 2023-05-21 at 10 21 17 AM

## Steps to test
1. Test that mutation is successful
2. Ensure component matches design as closely as possible

## What should reviewers focus on?
* Is there any extra code I should/can remove?
* Any File Name changes?

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Comment by wais-hundekar** [comment]
Just carrying over @jfdoming comment from the closed PR:
"Aside from my current comments, I'm also a little wary of the hard-coded month/weekday names. Dates are tricky and i18n will be difficult with this system (if we ever wanted to do that)! If it's too much work to migrate to a browser API for this I'm fine leaving that part though."

I would be interested in discussing the time tradeoff to implement this, and if the new PR changes would affect this in anyway.

File: frontend/src/components/user-management/student/AddClassroomModal.tsx [hunk 1]
```diff
@@ -151,13 +152,66 @@ const AddClassroomModal = ({
...
+                        "April",
+                        "May",
+                        "June",
+                        "July",
+                        "August",
+                        "September",
+                        "October",
+                        "November",
+                        "December",
+                      ], // length of 12
```
Comment (carissa-tang) [inline frontend/src/components/user-management/student/AddClassroomModal.tsx]: can we remove these comments?

File: frontend/src/components/user-management/student/AddClassroomModal.tsx [hunk 2]
```diff
@@ -151,13 +152,66 @@ const AddClassroomModal = ({
...
-                  <FormLabel color="blue.300">School Year</FormLabel>
-                  <Input
-                    defaultValue="2023"
-                    onChange={(e) => handleChange(e, "schoolYear")}
-                    placeholder="Type in School Year"
-                    type="number"
-                    value={watch("schoolYear")}
+                  <FormLabel color="blue.300">Start Date</FormLabel>
+                  <SingleDatepicker
+                    configs={{
```
Comment (carissa-tang) [inline frontend/src/components/user-management/student/AddClassroomModal.tsx]: can we move this to a new datepicker component in `common`?

File: frontend/src/components/user-management/student/AddClassroomModal.tsx [hunk 3]
```diff
@@ -151,13 +152,66 @@ const AddClassroomModal = ({
...
+                        "May",
+                        "June",
+                        "July",
+                        "August",
+                        "September",
+                        "October",
+                        "November",
+                        "December",
+                      ], // length of 12
+                      firstDayOfWeek: 1, // default is 0, the dayNames[0], which is Sunday if you don't specify your own dayNames,
```
Comment (carissa-tang) [inline frontend/src/components/user-management/student/AddClassroomModal.tsx]: wait this is design question but why does our calendar start on monday ?? 0_o

File: frontend/src/components/user-management/student/AddClassroomModal.tsx [hunk 4]
```diff
@@ -151,13 +152,66 @@ const AddClassroomModal = ({
...
                 <VStack align="left" direction="column" width="320px">
-                  <FormLabel color="blue.300">School Year</FormLabel>
-                  <Input
-                    defaultValue="2023"
-                    onChange={(e) => handleChange(e, "schoolYear")}
-                    placeholder="Type in School Year"
-                    type="number"
-                    value={watch("schoolYear")}
+                  <FormLabel color="blue.300">Start Date</FormLabel>
+                  <SingleDatepicker
```
Comment (carissa-tang) [inline frontend/src/components/user-management/student/AddClassroomModal.tsx]: can you add "Please choose a date" as a placeholder text?
image

File: backend/models/class.model.ts [hunk 5]
```diff
@@ -12,7 +12,7 @@ export interface Class extends Document {
   /** the name of the class */
   className: string;
   /** the school year of the class */
```
Comment (joyce-shi) [inline backend/models/class.model.ts]: could you update the comment? thanks!

File: frontend/src/components/user-management/student/AddClassroomModal.tsx [hunk 6]
```diff
@@ -151,13 +155,10 @@ const AddClassroomModal = ({
...
-                  <Input
-                    defaultValue="2023"
-                    onChange={(e) => handleChange(e, "schoolYear")}
-                    placeholder="Type in School Year"
-                    type="number"
-                    value={watch("schoolYear")}
+                  <FormLabel color="blue.300">Start Date</FormLabel>
+                  <DatePicker
+                    onChange={handleDateChange}
+                    value={watch("startDate")}
```
Comment (joyce-shi) [inline frontend/src/components/user-management/student/AddClassroomModal.tsx]: do we check somewhere that the `startDate` is non-empty and on or after the current date?

**Review by joyce-shi — COMMENTED** [review]
could you merge master? i think there might be some conflicts. then i will approve!

**Pull request closed without merging**